src/cli/generator/config.rs

```diff
@@ -90,18 +94,42 @@ pub struct Schema {
     pub query: Option<String>,
 }
 
-impl From<Preset> for config::transformer::Preset {
-    fn from(val: Preset) -> Self {
+impl TryFrom<Preset> for config::transformer::Preset {
+    type Error = ValidationError<String>;
+    fn try_from(val: Preset) -> Result<Self, Self::Error> {
         let mut preset = config::transformer::Preset::default();
+
         if let Some(merge_type) = val.merge_type {
+            if config::transformer::Preset::is_invalid_threshold(merge_type) {
+                return Valid::fail(format!(
+                    "Invalid threshold value ({:.2}). Allowed range is [0.0 - 1.0] inclusive.",
+                    merge_type
+                ))
+                .to_result();
+            }
             preset = preset.merge_type(merge_type);
         }
 
         if let Some(consolidate_url) = val.consolidate_url {
+            if config::transformer::Preset::is_invalid_threshold(consolidate_url) {
+                return Valid::fail(format!(
+                    "Invalid threshold value ({:.2}). Allowed range is [0.0 - 1.0] inclusive.",
+                    consolidate_url
+                ))
+                .to_result();
+            }
             preset = preset.consolidate_url(consolidate_url);
         }
 
-        preset
+        if let Some(use_better_names) = val.use_better_names {
+            preset = preset.use_better_names(use_better_names);
+        }
+
+        if let Some(tree_shake) = val.tree_shake {
+            preset = preset.tree_shake(tree_shake);
+        }
+
+        Ok(preset)
```

Comment:
Use `Valid` so that we can show multiple errors at the same time.

Reply:
I created a method on _Preset_ that return _Valid_ let me know if that's what you mean !